### PR TITLE
Add unit tests for color conversion functions

### LINKED ISSUE #678: RGB/HSV color conversion functions should be covered by unit tests
### Expected Behavior

The `rgbToHsv` and `hsvToRgb` color conversion functions in `EffectTransform.js` should be included in unit tests.

### Actual Behavior

Currently those functions are not tested. They're also not exported, which will be necessary to add tests.

### PR DESCRIPTION
**Pull request by adroitwhiz**
### Resolves

Resolves #678

### Proposed Changes

This PR moves `rgbToHsv` and `hsvToRgb` to their own file, and adds unit tests for them. The edge-case-finding part of my brain isn't firing on all cylinders today so feel free to suggest more useful tests.

### Reason for Changes

More tests = more better

### Test Coverage

Yes

### CODE REVIEW

File: test/unit/ColorConversionTests.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    message += `: found ${JSON.stringify(Array.from(found))}, wanted ${JSON.stringify(Array.from(wanted))}`;
+
+    // should always return another assert call, or
+    // this.pass(message) or this.fail(message, extra)
+    if (found.length !== wanted.length) {
+        return this.fail(message, extra);
+    }
+
+    for (let i = 0; i < found.length; i++) {
+        if (Math.abs(found[i] - wanted[i]) > 1e-3) {
```
Comment (fsih) [inline test/unit/ColorConversionTests.js:16]: What do you think of 1/256 or 1/360 for epsilon here?

File: test/unit/ColorConversionTests.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    return this.pass(message);
+    /* eslint-enable no-invalid-this */
+});
+
+test('RGB to HSV', t => {
+    const dst = [0, 0, 0];
+    t.colorsAlmostEqual(rgbToHsv([255, 255, 255], dst), [0, 0, 1], 'white');
+    t.colorsAlmostEqual(rgbToHsv([0, 0, 0], dst), [0, 0, 0], 'black');
```
Comment (fsih) [inline test/unit/ColorConversionTests.js:29]: Do we care about rgb(0,0,0) should equal hsv(0,1,0)?

File: test/unit/ColorConversionTests.js (hunk 1 above)
Comment (cwillisf) [inline test/unit/ColorConversionTests.js]: It'd be nice to use a tolerance that means something, rather than just an arbitrary `> 1e-3`. Maybe `>= 0.5 / 360`, so it'll reject any error that would cause variance in a hue value after rounding?

File: test/unit/ColorConversionTests.js (hunk 2 above)
Comment (fsih) [reply test/unit/ColorConversionTests.js:29]: I guess not since that's just a property of the test equals function

**Review by fsih — APPROVED** [review]
I think this is good to merge! Just going to leave it open for a bit for the discussion on the tolerance number

**Review by fsih — APPROVED** [review]
LG, thanks for tests!
